generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs

```diff
@@ -78,14 +80,19 @@ public class <%= entityClass %>RepositoryWithBagRelationshipsImpl implements <%=
     }
 
     List<<%= persistClass %>> fetch<%= relationship.relationshipNameCapitalizedPlural %>(List<<%= persistClass %>> <%= entityInstancePlural %>) {
-        return entityManager
+        HashMap<Object, Integer> order = new HashMap<Object, Integer>();
```

Comment:
Could you add the order in a dedicated private method (or at least the sorting stuff at the bottom of the method)

Reply:
Does the createQuery function changes the original array?